plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -1348,18 +1337,15 @@ read_only::get_activated_protocol_features( const read_only::get_activated_proto
       auto& activation_ordinal_value   = mvo["activation_ordinal"];
       auto& activation_block_num_value = mvo["activation_block_num"];
 
-      auto cur_time = fc::time_point::now();
-      for( unsigned int count = 0;
-           cur_time <= params_deadline && count < params.limit && itr != end_itr;
-           ++itr, cur_time = fc::time_point::now() )
-      {
-         FC_CHECK_DEADLINE(deadline);
+      // activated protocol features unlikely to ever reach max_return_items
+      for( unsigned int count = 0; count < params.limit && itr != end_itr; ++itr, ++count ) {
          const auto& conv_itr = convert_iterator( itr );
          activation_ordinal_value   = conv_itr.activation_ordinal();
          activation_block_num_value = conv_itr.activation_block_num();
 
          result.activated_protocol_features.emplace_back( conv_itr->to_variant( false, &mvo ) );
-         ++count;
+         if (fc::time_point::now() >= params_deadline)
+            break;
```

Comment:
Should we even check the deadline in this case, since this is an atomic request and we said the time limit applied only to non-atomic requests.

Reply:
This is non-atomic because the number of activate protocol features is unbounded. However, unlikely to be very large ever. So yeah, I think we should just remove. If we do remove then we will ignore the user's requested time limit. Should we also ignore the users requested `params.limit` ?

Reply:
Yes, I think we should ignore the user's requested time limit. I would consider it atomic because the user cannot specify a range, so if it is less than the configured `http_max_response_time` the user would not be able to get the info.